libs/SalesforceSDKCommon/SalesforceSDKCommon/Classes/Keychain/KeychainManager.swift

```diff
@@ -0,0 +1,272 @@
+public class KeychainHelper: NSObject {
+    
+    /// Read a value from the keychain.
+    /// - Parameters:
+    ///   - service: Identifier to use for keychain service key.
+    ///   - account: Identifier to use for keychain account key.
+    /// - Returns: KeychainResult
+    @objc public class func read(service: String, account: String?) -> KeychainResult {
+        let keychainManager = KeychainItemManager(service: service, account: account)
+        return keychainManager.getValue()
+    }
+    
+    
+    /// Create an item in the keychain if not present.
+    /// - Parameters:
+    ///   - service: Identifier to use for keychain service key.
+    ///   - account: Identifier to use for keychain account key.
+    /// - Returns: KeychainResult
+    @objc public class func createIfNotPresent(service: String, account: String?) -> KeychainResult {
+        let keychainManager = KeychainItemManager(service: service, account: account)
+        var keychainResult =  keychainManager.getValue()
+        if !keychainResult.success && keychainResult.status == errSecItemNotFound {
+            keychainResult =  keychainManager.addEmptyValue()
+        }
+        return keychainResult
+    }
+    
+    /// Write or Update an item in the keychain if not present.
+    /// - Parameters:
+    ///   - service: Identifier to use for keychain service key.
+    ///   - data: Data to write
+    ///   - account: Identifier to use for keychain account key.
+    /// - Returns: KeychainResult
+    @objc public class func write(service: String, data: Data, account: String?) -> KeychainResult  {
+        let keychainManager = KeychainItemManager(service: service, account: account)
+        return keychainManager.setValue(data)
+    }
+    
+    /// If an item is found remove it and then add an empty entry i.e. data is nil.
+    /// - Parameters:
+    ///   - service: Identifier to use for keychain service key.
+    ///   - account: Identifier to use for keychain account key.
+    /// - Returns: KeychainResult
+    @objc public class func reset(service: String, account: String?) -> KeychainResult {
+        let keychainManager = KeychainItemManager(service: service, account: account)
+        var keychainResult = keychainManager.getValue()
+        if keychainResult.success == true, keychainManager.removeValue().success {
+            keychainResult = keychainManager.addEmptyValue()
+        }
+        return keychainResult
+    }
+    
+    
+    /// Remove an item from the keychain.
+    /// - Parameters:
+    ///   - service: Identifier to use for keychain service key.
+    ///   - account: Identifier to use for keychain account key.
+    /// - Returns: KeychainResult
+    @objc public class func remove(service: String, account: String?) -> KeychainResult {
+        let keychainManager = KeychainItemManager(service: service, account: account)
+        var keychainResult = keychainManager.getValue()
+        if keychainResult.success == true {
```

Comment:
Same comment here about the comparison to boolean literal

Reply:
verbose. Ill fix it.